Source/SwiftLintFramework/Rules/ValidDocsRule.swift

```diff
@@ -51,7 +51,9 @@ func declarationReturns(declaration: String, kind: SwiftDeclarationKind? = nil)
     guard let outsideBracesMatch = matchOutsideBraces(declaration) else {
         return false
     }
-
+    if declaration.containsString("init(") {
```

Comment:
@freak4pc You are right, should've  considered the constant string more carefully.
* I'm not that familiar with the project to be honest, and I wasn't aware that the `declaration`  string would be something like: `public convince init()` or plain `init()`, that's why I didn't go for the `hasPrefix`, but I see your point and I think a leading space could help!
* I decided against the `guard` usage instead of `if` because the guidelines I'm following, prohibits using guards for case testing and this is clearly testing against different scenarios, but, if SwiftLint suggests I should use `guard`, by all means!
* The very next line of code uses the same technique for checking if a function returns (by checking the existence of `->`), granted, that is very much unique in the swift grammar, but then, so is `init()`, don't you agree?